server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java

```diff
@@ -150,19 +150,6 @@ public static Query refExistsQuery(Reference ref, Context context) {
             if (fieldType != null && !fieldType.omitNorms()) {
                 return new FieldExistsQuery(field);
             }
-            if (ArrayType.unnest(ref.valueType()) instanceof ObjectType objType) {
```

Comment:
Arrays are covered above, plain object is covered by 
`if (fieldType == null || fieldType.indexOptions() == IndexOptions.NONE && !fieldType.stored()) { `below, we don't need this code.

Reply:
The code below optimized `OBJ IS NULL` to utilize the indices of the _inner_ columns, to figure out if any of the child columns are present. If the child columns exist it implies that the object can't be null.

Removing this would lead to a performance regression?